### DISCUSSION TITLE
Get paths from cyclic MultiDiGraph

### DISCUSSION THREAD

**Discussion by bdrum** [networkx/networkx#6718] [Q&A]
Dear experts,

First of all many thank for networkx. It's absolutely brilliant!

Could you please help me with creating a travers algorithm for my graph.

I've MultiDigraph where edges has duration property. 
and I have input values for algorithms such as start_period, max_period and starting node.

The task is to reduce graph to one that doesn't contain unreached nodes by duration.
The reason to stop iterate a is when start_period + sum of edge durations in path should be greater than max_period or 
if such edge already in the path.

I've an idea to traverse graph with bfs and keep all passed paths, but have faced with cycle problem.

For example:
![image](https://github.com/networkx/networkx/assets/16210997/09b2de6c-7853-4e64-a1cb-4179b1f55c44)

Image has graph with 6 nodes and some number of edges.

As you can see it has few cycles:

1 - 2 - 3 - 1
1 - 5 - 6 - 2 - 3 - 1

Could you please suggest any idea how to solve this problem?
Perhaps I've to simplify the graph or some algorithms exists for such cases that I didn't know...

**Comment by dschult** [comment]
How would you like to handle cycles?  Would it be ok to stop checking for paths that include the complete cycle since you will find those nodes via shorter paths that don't include the cycle?  If so, you can use the bfs because it won't repeat a cycle multiple times. 

But maybe better than that is to compute `nx.single_source_shortest_path_length(G, source, weight="duration", cutoff=max_period)`.  But I may be missing something.

**Reply by bdrum** [reply]
Thanks!

Hm, sorry I forgot to mention the condition for such cases.
Let's say that I have a number for each edge shows how many times it could be passed.

Yes, I thought about shortest path, but in my case it is possible to use path if it is not shortest, but passed duration condition,
shortest paths in your example it will be all paths that satisfy condition for duration or paths with minimal duration? I suggest the first one...
In this case it's ok, thanks.
Then the question is there a possibility to add heuristic to handle cycles something like AStar or how to do it better?

**Reply by bdrum** [reply]
@dschult 
Hm, I've tried to test your suggestion and have seen that nx.single_source_shortest_path_length doesn't have weighted version.
Is it correct or I'm missing something?

**Comment by bdrum** [comment]
Let me please provide more detailes about my graph.
I've MultiDigraph has 1.5k nodes and 100k of edges.
Avg out degree ~= avg in degree = 60.

As I already mentioned it has cycles.

I've implemented the algorithm of traverse pathes via bfs, but in case of rising duration  for one day it gives me multiplication factor 60, so if I want to search all pathes satisfy my condition for 10 day, I've $60^{10}$ variants.

Is there solution except reorganizing graph, what do you think?
